modules/generator/processor/servicegraphs/servicegraphs.go

```diff
@@ -50,26 +62,30 @@ type processor struct {
 	collectCh chan string
 	closeCh   chan struct{}
 
-	serviceGraphRequestTotal           *prometheus.CounterVec
-	serviceGraphRequestFailedTotal     *prometheus.CounterVec
-	serviceGraphRequestServerHistogram *prometheus.HistogramVec
-	serviceGraphRequestClientHistogram *prometheus.HistogramVec
-	serviceGraphUnpairedSpansTotal     *prometheus.CounterVec
-	serviceGraphDroppedSpansTotal      *prometheus.CounterVec
+	serviceGraphRequestTotal                  registry.Counter
+	serviceGraphRequestFailedTotal            registry.Counter
+	serviceGraphRequestServerSecondsHistogram registry.Histogram
+	serviceGraphRequestClientSecondsHistogram registry.Histogram
+	//serviceGraphUnpairedSpansTotal            registry.Counter
+	//serviceGraphDroppedSpansTotal             registry.Counter
```

Comment:
These metrics weren't updated before, so I don't know if we should keep them around. I'll check if Grafana actually uses these in the service graphs view.

We already expose these metrics as metrics from Tempo:
 - `metrics_generator_processor_service_graphs_dropped_spans`
 - `metrics_generator_processor_service_graphs_unpaired_edges`

So a Tempo operator can see these stats already.

Reply:
We probably need to have these per tenant so that individual tenants can have a feeling for the quality of their metrics. Alternatively, instead of exposing these directly, it might be better to use the new `tempo_warnings_total` metric with a specific `reason` label for these situations.

Reply:
We currently keep track of dropped spans and unpaired edges per tenant. So we already collect the data, I think it will just be a matter of exposing them to the end-user.
`tempo_warnings_total` looks interesting, I'll take a look at what warnings we want to share from the metrics-generator in general.